src/cpp/src/vlm_pipeline.cpp

```diff
@@ -341,7 +282,137 @@ ov::Tensor merge_text_and_image_embeddings_llava(
+    if (m_vlm_config.model_type == VLMModelType::LLAVA) {
+        request->update_processed_tokens_num(request->get_prompt_len());
+    }
+
+    language.get_tensor("inputs_embeds").set_shape({BATCH_SIZE, 1, m_vlm_config.hidden_size});
+    language.get_tensor("position_ids").set_shape({ BATCH_SIZE, 1 });
+
+    while (!request->has_finished()) {
+        request->schedule_tokens(1);
+        size_t num_sequences = request->num_running_seqs();
+        size_t total_num_tokens = request->get_num_scheduled_tokens() * num_sequences;
+
+        ov::Tensor
+            input_ids(ov::element::i64, {total_num_tokens, 1}),
+            position_ids(ov::element::i64, {total_num_tokens, 1}),
+            beam_idx(ov::element::i32, { total_num_tokens });
+
+        int64_t
+            * input_ids_data = input_ids.data<int64_t>(),
+            * position_ids_data = position_ids.data<int64_t>();
+
+        size_t num_scheduled_tokens = request->get_num_scheduled_tokens();
+        size_t group_position_id = request->get_num_processed_tokens();
+        for (Sequence::Ptr& sequence : request->get_running_sequences()) {
+            for (size_t token_id = 0, position_id = group_position_id; token_id < num_scheduled_tokens; ++token_id, ++position_id) {
+                // compute token for current sequence
+                input_ids_data[token_id] = position_id < request->get_prompt_len() ?
+                    request->get_prompt_ids()[position_id] :
+                    sequence->get_generated_ids()[position_id - request->get_prompt_len()];
+
+                position_ids_data[token_id] = position_id;
+            }
+            // apply strides to shift to a next sequence
+            input_ids_data += num_scheduled_tokens;
+            position_ids_data += num_scheduled_tokens;
+        }
+
+        embedding.set_input_tensor(input_ids);
+
+        embedding.infer();
+        const ov::Tensor& embed_prompt_tensor = embedding.get_output_tensor();
+        float* embed_data = embed_prompt_tensor.data<float>();
+        for (auto idx = 0; idx < embed_prompt_tensor.get_size(); idx++) {
+            embed_data[idx] = embed_data[idx] * m_vlm_config.scale_emb;
+        }
+
+        language.set_tensor("inputs_embeds", embed_prompt_tensor);
+
+        language.get_tensor("attention_mask").set_shape({ total_num_tokens, language.get_tensor("attention_mask").get_shape()[1] + 1 });
+        std::fill_n(language.get_tensor("attention_mask").data<int64_t>(), language.get_tensor("attention_mask").get_size(), 1);
+
+        language.set_tensor("position_ids", position_ids);
+
+        std::vector<int32_t> beam_idxs = sampler.get_beam_idxs(request->get_request_id());
+        int32_t *beam_idx_data = beam_idx.data<int32_t>();
+        if (total_num_tokens > beam_idxs.size()) {
+            std::fill_n(beam_idx_data, total_num_tokens, 0);
+        } else {
+            copy(beam_idxs.begin(), beam_idxs.end(), beam_idx_data);
+        }
```

Comment:
do we need to handle such special case with sampling ? I supposed we can always copy `beam_idxs` as after first `sample()` call it always provides proper `beam_idxs`, does not it?

Reply:
in case of multinomial with num_return_sequences > 1 sampler will provide { 0 }, but we need to { 0 * num_return_sequences  }

Reply:
do you think it can be reflected in `get_beam_idx` method? so it return proper values even for parallel sampling

Reply:
yes, like that https://github.com/openvinotoolkit/openvino.genai/blob/c3fe8d4098a949413cbcd6aec451ecfff8c8d237/src/cpp/src/sampler.cpp#L604